lib/Image/Info/JPEG2000.pm

```diff
@@ -6,21 +6,30 @@ sub ImageInfo {
     return image_info(@_);
 }
 
+sub process_file {
+    my($info, $fh) = @_;
+    my $result = image_info($fh);
+    foreach my $key ( keys %$result ) {
+        $info->push_info(0, $key => $$result{$key});
+    }
+}
+
 sub image_info {
     my ( $filename ) = @_;
-    my $fh = new IO::File $filename;
+    my $fh = ref($filename) ? $filename : new IO::File $filename;
     my ( $tests, $data ) = BoxValidator->new("JP2", $fh)->validate();
     my $result = {
         width => $$data{'jp2HeaderBox/imageHeaderBox/width'},
         height => $$data{'jp2HeaderBox/imageHeaderBox/height'},
         layers => $$data{'contiguousCodestreamBox/cod/layers'},
         levels => $$data{'contiguousCodestreamBox/cod/levels'},
     };
-    if ( $$data{'colourSpecificationBox/enumCS'} == 16 ) {
+
+    if ( $$data{'jp2HeaderBox/colourSpecificationBox/enumCS'} == 16 ) {
         $$result{ColorSpace} = qq{sRGB};
-    } elsif ( $$data{'colourSpecificationBox/enumCS'} == 17 ) {
+    } elsif ( $$data{'jp2HeaderBox/colourSpecificationBox/enumCS'} == 17 ) {
         $$result{ColorSpace} = qq{sLUM};
-    } elsif ( $$data{'colourSpecificationBox/enumCS'} == 18 ) {
+    } elsif ( $$data{'jp2HeaderBox/colourSpecificationBox/enumCS'} == 18 ) {
```

Comment:
It wasn't clear to me if this is your code or code copied from `Image::Info` (at least without going digging through `Image::Info`, but FWIW there shouldn't be any JPEG2000 images in the repository with color space `sYCC`)

Reply:
This is mostly a port of the BoxValidator from jpylyzer.